finagle-thrift/src/main/scala/com/twitter/finagle/thrift/TTwitterServerFilter.scala

```diff
@@ -59,11 +60,10 @@ private[finagle] class TTwitterServerFilter(
               env, Buf.ByteArray.Owned(c.getKey()), Buf.ByteArray.Owned(c.getValue()))
           }
         }
-
-        Trace.recordRpc({
-          val msg = new InputBuffer(request_, protocolFactory)().readMessageBegin()
-          msg.name
-        })
+        val msg = new InputBuffer(request_, protocolFactory)().readMessageBegin()
+        Trace.recordRpc(msg.name)
+        val rpcStatsReciver = ServerStatsReceiver.scope(serviceName).scope(msg.name)
```

Comment:
for what it's worth, scoping new stats receivers, and creating new counters should never be done on the hot path unless it's impossible to know the name of the counter until runtime.  since we know the names in advance, we can do better.